server/resource_server/service/service_test.go

```diff
@@ -64,8 +96,18 @@ func (m *mockRepository) ProfileUser(userID uint) (models.User, []models.UserMet
 	return models.User{}, []models.UserMetadata{}, fmt.Errorf("User not found")
 }
 
-func (m *mockRepository) VerifyEmail(userID uint) error {
-	return nil
+func (m *mockRepository) SaveProofOfEmailVerification(
+	userID uint, verificationCode string, proof string,
+) error {
+	return m.saveProofOfEmailVerification(userID, verificationCode, proof)
```

Comment:
On lines 506,  & 532, you overwrite the function on the mockRepo so why the recursive implementation here? Where `m.SaveProofOfEmailVerification` calls `m.SaveProofOfEmailVerification()` on itself? Why not simply `return nil`?

Reply:
This is a pattern to inject a mock method implementation:
1. We create a field in the mock struct for each method that we want to mock
2. We call this function field, in this example ```m.saveProofOfEmailVerification```, in the interface method that is overridden by the struct, in this case ```SaveProofOfEmailVerification```
3. We pass the concrete implementation of method as the struct field when creating an instance of the mock struct. In this case, we do it in lines 523 and 542 within tests to inject a mock implementation of the repository method ```SaveProofOfEmailVerification```.